lib/Target/Jaxite/JaxiteEmitter.cpp

```diff
@@ -229,6 +233,54 @@ LogicalResult JaxiteEmitter::printOperation(tensor::FromElementsOp op) {
   return success();
 }
 
+// Loading variables.
+// Example: temp_nodes[idx] = input[i]
+LogicalResult JaxiteEmitter::printOperation(memref::LoadOp op) {
+  emitAssignPrefix(op.getResult());
+  os << variableNames->getNameForValue(op.getMemref());
+  os << bracketEnclosedValues(op.getIndices(), [&](Value value) {
+    SmallString<16> idx_str;
+    dyn_cast<IntegerAttr>(
+        dyn_cast<arith::ConstantOp>(value.getDefiningOp()).getValue())
+        .getValue()
+        .toStringUnsigned(idx_str);
+    return idx_str.c_str();
+  });
+  os << "\n";
+  return success();
+}
+
+// Allocating output ciphertexts.
+// Example: temp_nodes[idx] = [types.LweCiphertext]*size
+LogicalResult JaxiteEmitter::printOperation(memref::AllocOp op) {
+  emitAssignPrefix(op.getResult());
+  os << "[";
+  if (failed(emitType(op.getMemref().getType().getElementType()))) {
+    return op.emitOpError() << "Failed to get memref element type";
+  }
+  os << "] * " << op.getMemref().getType().getNumElements();
+  os << "\n";
+  return success();
+}
+
+// Assuming StoreOp is only used while storing results.
+// Example: temp_nodes[result_idx][idx] = temp_nodes[i]
+LogicalResult JaxiteEmitter::printOperation(memref::StoreOp op) {
+  os << "temp_nodes[" << variableNames->getIntForValue(op.getMemref()) << "]";
+  os << bracketEnclosedValues(op.getIndices(), [&](Value value) {
```

Comment:
see my comment above, but maybe memref::allocop needs to be changed then? 

i think it might not end up showing up for your tests since python typing is pretty loose and you're always allocating at least as many elements as were in the first dimension of the memref

Reply:
Yup! memref::storeop and memref::allocop need be in sync. You are right, memref::allocop is allocating the "flattened" array of ciphertexts and [this](memref::StoreOp) one is storing at unflattened indices and that's inconsistent. Updated memref::allocop to reflect the shape of the memref.  This should be consistent now.